### ISSUE TITLE
Usage of more efficient Sandia Modules?

### ISSUE THREAD

**Issue by k-rieck** [pvlib/pvlib-python#1929]
Dear PVLib Team,

we would like to use more efficient PV modules with efficiencies of 21-22% as they are represented in your CEC module database (e.g. "Trina Solar TSM-410DE09")  
Your pvlib functions (e.g pvlib.pvsystem.samp) seem to work only with Sandia modules which seem to have lower efficiencies.
Is there a way to use the pvlib functions on more efficient modules?

Many thanks in advance!

**Comment by kandersolar** [comment]
Yes, these modules can be simulated with pvlib.  To use parameters taken from the CEC module database, you must use the CEC performance model, which is available in pvlib with `pvlib.pvsystem.calcparams_cec` and `pvlib.pvsystem.singlediode`.  `ModelChain` can use CEC module parameters as well.

- https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.pvsystem.calcparams_cec.html
- https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.pvsystem.singlediode.html
- https://pvlib-python.readthedocs.io/en/stable/user_guide/pvsystem.html#module-and-inverter-parameters

**Comment by k-rieck** [comment]
Thanks a lot! We will check it out

**Comment by k-rieck** [comment]
Hi, I tried to follow the example in the link https://pvlib-python.readthedocs.io/en/stable/gallery/iv-modeling/plot_singlediode.html#sphx-glr-gallery-iv-modeling-plot-singlediode-py.
Instead of using "pvsystem.calcparams_desoto" I tried "pvlib.pvsystem.calcparams_cec".
Is there a way to calculate the effective irradiance like in "pvlib.pvsystem.sapm_effective_irradiance" for the cec modules?
Thanks a lot!

**Comment by cwhanse** [comment]
Effective irradiance is POA global irradiance reduced by reflections of the direct component and adjusted for spectrum.

You can use the SAPM bundled approach, but that requires coefficients you probably don't have.

You can use functions in `pvlib.iam` to get the reflection loss multiplier, and functions in `pvlib.spectrum` to get the spectral adjustments.

This process could really use a good example in the Gallery, and maybe, a convenience function in `pvlib.pvsystem.`
